reVX/pras/rev_reeds_pras.py

```diff
@@ -0,0 +1,690 @@
+    overwrite exisiting pras file with new generation profiles. This class
+    takes as input the plexos nodes meta data (lat/lon or shape files), rev
+    supply curve table, and reeds buildout table (specifying which rev sc
+    points were built and at what capacity). The class will build power
+    profiles for each supply curve point and then aggregate the sc point
+    profiles to the nearest neighbor pras node (if pras nodes are defined by
+    lat/lon) or the shape intersect pras node (if pras nodes are defined by
+    shape file). The corresponding node power profiles in the pras output file
+    will be overwritten with these new aggregated generation profiles.
+
+    Examples
+    --------
+
+    >>> from reVX.pras.rev_reeds_pras import PrasAggregation
+    >>>
+    >>> node_shape_file = ...
+    >>> rev_supply_curve_file = ...
+    >>> reeds_build_file = ...
+    >>> capacity_factor_file = ...
+    >>> pras_file = ...
+    >>>
+    >>> PrasAggregation.run(node_shape_file, rev_supply_curve_file,
+                            reeds_build_file, capacity_factor_file,
+                            pras_file, build_year=2047, tech_type='upv')
+
+    This will replace all the pras generation profiles with upv tech in the
+    file pras_file.pras.h5 with aggregated generation profiles from cf_path.h5
+
+    >>> reeds_build_files = [...]
+    >>> capacity_factor_files = [...]
+    >>> tech_types = ['upv', 'dupv', 'wind-ons', 'wind-ofs']
+    >>>
+    >>> PrasAggregation.run(node_shape_file, rev_supply_curve_file,
+                            reeds_build_files, capacity_factor_files,
+                            pras_file, build_year=2047,
+                            tech_type=tech_types)
+
+    This will run the same routine but for each tech type in tech_types. The
+    files in reeds_build_files and capacity_factor_files have to correspond to
+    the tech type in the tech_types list.
+    """
+    def __init__(self, plexos_nodes, rev_sc, reeds_build, cf_fpath, pras_file,
+                 load_path=None, forecast_fpath=None, build_year=2050,
+                 plexos_columns=None, force_full_build=False,
+                 force_shape_map=False, plant_name_col=None, tech_tag=None,
+                 res_class=None, tech_type=None, timezone='UTC',
+                 dset_tag=None, max_workers=None):
+        """
+        Parameters
+        ----------
+        plexos_nodes : str | pd.DataFrame
+            Plexos node meta data including gid, latitude, longitude, voltage.
+            Or file path to .csv containing plexos node meta data, or a file
+            path to a .shp file that contains plexos nodes defined as shapes.
+        rev_sc : str | pd.DataFrame
+            reV supply curve results table including SC gid, latitude,
+            longitude, res_gids, gid_counts. Or file path to reV supply
+            curve table.
+        reeds_build : str | pd.DataFrame
```

Comment:
if we're going to require that this include the reeds zone for this module, we should specify here